packages/CodingStyle/src/Rector/ClassMethod/AnnotateThrowables.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodingStyle\Rector\ClassMethod;
+
+use PhpParser\Comment\Doc;
+use PhpParser\Node;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Throw_;
+use Rector\Rector\AbstractRector;
+use Rector\RectorDefinition\CodeSample;
+use Rector\RectorDefinition\RectorDefinition;
+
+/**
+ * Adds "throws" DocBlock to methods.
+ */
+final class AnnotateThrowables extends AbstractRector
+{
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes() : array
+    {
+        return [Throw_::class];
+    }
+
+    /**
+     * @param Node|Throw_ $node
+     *
+     * @return Node|null
+     */
+    public function refactor(Node $node) : ?Node
+    {
+        if ($this->isThrowableAnnotated($node)) {
+            return null;
+        }
+
+        return $this->annotateMethod($node);
+    }
+
+    /**
+     * @param Throw_ $node
+     *
+     * @return bool
+     */
+    private function isThrowableAnnotated(Throw_ $node) : bool
+    {
+        $method = $node->getAttribute('methodNode');
+        $throwableParts = $node->expr->class->parts;
+        $docComment = $method->getDocComment();
+        $comments = $method->getComments();
+
+        if (null === $docComment && empty($comments)) {
+            return false;
+        }
+
+        return $this->isFQNAnnotated($docComment, $comments, $throwableParts);
+    }
+
+    /**
+     * @param Doc $docComment
+     * @param array<Doc> $comments
+     * @param array<string> $throwableParts
+     *
+     * @return bool
+     */
+    private function isFQNAnnotated(Doc $docComment, array $comments, array $throwableParts):bool
+    {
+        $FQN = $this->buildFQN($throwableParts);
+        $pattern = $this->buildThrowsDocComment($FQN);
+
+        if ($this->isDocCommentAlreadyPresent($docComment, $pattern)) {
+            return  true;
+        }
+
+        foreach ($comments as $comment) {
+            if ($this->isDocCommentAlreadyPresent($comment, $pattern)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param Doc $docComment
+     * @param string                 $pattern
+     *
+     * @return bool
+     */
+    private function isDocCommentAlreadyPresent(Doc $docComment, string $pattern):bool
+    {
+        return false !== strpos($docComment->getText(), $pattern);
+    }
+
+    /**
+     * @param string $FQNOrThrowableName
+     *
+     * @return string
+     */
+    private function buildThrowsDocComment(string $FQNOrThrowableName):string
+    {
+        return sprintf('@throws %s', $FQNOrThrowableName);
+    }
+
+    /**
+     * @param array $throwableParts
+     *
+     * @return string
+     */
+    private function buildFQN(array $throwableParts):string
+    {
+        return '\\' . implode('\\', $throwableParts);
+    }
+
+    /**
+     * @param Throw_ $node
+     *
+     * @return Throw_
+     */
+    private function annotateMethod(Throw_ $node):Throw_
```

Comment:
You can use object instead:

```php
$method->getAttribute(AttributeKey::PHP_DOC_INFO)
```

Also, no need to return node, since it holds the reference to the object.
Just return void